aiozk/recipes/base_lock.py

```diff
@@ -15,7 +15,18 @@ class BaseLock(SequentialRecipe):
         if timeout is not None:
             time_limit = time.time() + timeout
 
-        await self.create_unique_znode(znode_label)
+        try:
+            await self.create_unique_znode(znode_label)
+        except Exception as ex:
+            log.debug('unknown exception: %s', ex)
+            while True:
+                try:
+                    await self.delete_garbage_znodes(znode_label)
```

Comment:
I also feel not easy about loop without any cooldown. Probably it isn't urgent and can be handled later or we can wait for connected state before deleting garbage znodes